sdlib/d/gc/hpd.d

```diff
@@ -111,6 +111,46 @@ public:
 		return bestIndex;
 	}
 
+	bool reserve(uint index, uint pages) {
+		assert(pages > 0 && pages <= PageCount, "Invalid number of pages!");
+		assert(index <= PageCount - pages, "Invalid index!");
+
+		if (pages > longestFreeRange) {
+			return false;
+		}
+
+		uint freePos, freeLength;
+		allocatedPages.nextFreeRange(index, freePos, freeLength);
+
+		if ((freePos != index) || (freeLength < pages)) {
+			return false;
+		}
+
+		allocatedPages.setRange(index, pages);
+		usedCount += pages;
+
+		if (freeLength != longestFreeRange) {
+			return true;
+		}
+
+		// If allocated from the longest free range, compute the new one:
```

Comment:
This comment belong on the if above (and needs to be reformulated somewhat). We already made the choice here.